hyperon_das_atomdb/adapters/ram_only.py

```diff
@@ -372,57 +353,81 @@ def _update_atom_indexes(self, documents: Iterable[dict[str, Any]], **kwargs) ->
         for document in documents:
             self._update_index(document, **kwargs)
 
-    def _update_index(self, atom: AtomT, **kwargs) -> None:
```

Comment:
I just brokedown the blocks `if` and `else` into separated/dedicated sub-functions.